opentelemetry/proto/common/v1/common.proto

```diff
@@ -21,30 +21,37 @@ option java_package = "io.opentelemetry.proto.common.v1";
 option java_outer_classname = "CommonProto";
 option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1";
 
+// AnyValue is used to represent any type of attribute value. AnyValue may contain a
+// primitive value such as a string or integer or it may contain an arbitrary nested
+// object containing arrays, key-value lists and primitives.
+message AnyValue {
+  // The value is one of the listed fields. It is valid for all values to be unspecified
+  // in which case this AnyValue is considered to be "null".
+  oneof value {
+    string string_value = 1;
+    bool bool_value = 2;
+    int64 int_value = 3;
+    double double_value = 4;
+    ArrayValue array_values = 5;
+    AttributeKeyValueList kvlist_values = 6;
```

Comment:
There's a little inconsistency here. Can we use "array_value" instead of "array_values"? The array is the value and the type is "ArrayValue".

Can we use a type named "MapValue" or "DictionaryValue" instead of "AttributeKeyValueList"? Can we name the corresponding field "map_value" or "dictionary_value"? Use of "map" or "dictionary" implies that the keys will be unique, whereas "list" does not.

Reply:
Renamed to `array_value` and `kvlist_value`.

I am not sure about using the term `map` here. From protocols perspective this is a list of elements. See also trace.proto `Span.attributes` where we describe it as "a collection of key/value pairs" and avoid calling it a map.

The proto here avoids taking a stance and specifying whether it is a map or multimap (both are possible for a key-value pair list). I have a slight preference to keep it this way to be aligned with how `Span.attributes` are defined.

Thoughts?

Reply:
Sounds good. thanks.